temporian/core/operators/arithmetic.py

```diff
@@ -0,0 +1,186 @@
+
+        if event_1.sampling() != event_2.sampling():
+            raise ValueError("event_1 and event_2 must have same sampling.")
+
+        if len(event_1.features()) != len(event_2.features()):
+            raise ValueError(
+                "event_1 and event_2 must have same number of features."
+            )
+
+        sampling = event_1.sampling()
+
+        prefix = ArithmeticOperation.prefix(operation)
+
+        # outputs
+        output_features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=f"{prefix}_{event_1_f.name()}_{event_2_f.name()}",
+                dtype=event_1_f.dtype(),
+                sampling=sampling,
+                creator=self,
+            )
+            for event_1_f, event_2_f in zip(
+                event_1.features(), event_2.features()
+            )
+        ]
+
+        self.add_output(
+            "event",
+            Event(
+                features=output_features,
+                sampling=sampling,
+                creator=self,
+            ),
+        )
+        self.check()
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="ARITHMETIC",
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="operation",
+                    type=pb.OperatorDef.Attribute.Type.STRING,
+                    is_optional=False,
+                ),
+                pb.OperatorDef.Attribute(
+                    key="resolution",
+                    type=pb.OperatorDef.Attribute.Type.STRING,
+                    is_optional=False,
+                ),
+            ],
+            inputs=[
+                pb.OperatorDef.Input(key="event_1"),
+                pb.OperatorDef.Input(key="event_2"),
+            ],
+            outputs=[pb.OperatorDef.Output(key="event")],
+        )
+
+
+operator_lib.register_operator(ArithmeticOperator)
+
+
+def sum(
+    event_1: Event,
+    event_2: Event,
+    resolution: Resolution = Resolution.PER_FEATURE_IDX,
+) -> Event:
+    return ArithmeticOperator(
+        event_1=event_1,
+        event_2=event_2,
+        operation=ArithmeticOperation.ADDITION,
+        resolution=resolution,
+    ).outputs()["event"]
+
+
+def substraction(
+    event_1: Event,
+    event_2: Event,
+    resolution: Resolution = Resolution.PER_FEATURE_IDX,
+) -> Event:
+    return ArithmeticOperator(
+        event_1=event_1,
+        event_2=event_2,
+        operation=ArithmeticOperation.SUBTRACTION,
+        resolution=resolution,
+    ).outputs()["event"]
+
+
+def multiplication(
+    event_1: Event,
+    event_2: Event,
+    resolution: Resolution = Resolution.PER_FEATURE_IDX,
+) -> Event:
+    return ArithmeticOperator(
+        event_1=event_1,
+        event_2=event_2,
+        operation=ArithmeticOperation.MULTIPLICATION,
+        resolution=resolution,
+    ).outputs()["event"]
+
+
+def division(
```

Comment:
I think we should make explicit which event is the numerator and which is the denominator.

Reply:
💯